docs/help-pages.rst

```diff
@@ -0,0 +1,242 @@
+Help Pages
+===================
+
+.. currentmodule:: click
+
+Click makes it very easy to document your command line tools. For most things Click automatically generates help pages for you. By design the text is customizable, but the layout is not.
+
+Help Texts
+----------
+
+Commands and options accept help arguments. For commands, the docstring of the function is automatically used if provided.
+
+Simple example:
+
+.. click:example::
+
+    @click.command()
+    @click.option('--count', default=1, help='number of greetings')
+    @click.option('--name', help='a name')
+    def hello(count, name=False):
```

Comment:
```suggestion
    def hello(count: int, name: str):
```

`name=False` is incorrect, no?

Let's also add typing here while we're at it.

Reply:
Seeing that this also changed from parameter, to option. I think we should not change this to an option. The goal here is to showcase how these have help pages. Highlighting parameters don't have help pages is important.

This is regardless of whether or not that's a good state right now. I think it should also get documentation, but that's outside of the scope of this PR.

Reply:
Done